catalog/rocks/catalog.go

```diff
@@ -0,0 +1,338 @@
+package rocks
+
+import (
+	"context"
+	"time"
+)
+
+// Basic Types
+
+// DiffType represents a changed state for a given entry (added, removed, changed, conflict)
+type DiffType uint8
+
+const (
+	DiffTypeAdded DiffType = iota
+	DiffTypeRemoved
+	DiffTypeChanged
+	DiffTypeConflict
+)
+
+// function/methods receiving the following basic types could assume they passed validation
+type (
+	// StorageNamespace is the URI to the storage location
+	StorageNamespace string
+
+	// RepositoryID is an identifier for a repo
```

Comment:
```suggestion
	// RepositoryID is an identifier for a repo
```
I think we enforce allowed characters here.